docs/central-upgrade.rst

```diff
@@ -25,65 +26,342 @@ Upgrade notes
+
+Version-specific upgrade instructions
+--------------------------------------
+
+.. _central-upgrade-2023.2:
+
+Upgrading to Central v2023.2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is *critical infrastructure upgrade*. It updates many aspects of the infrastructure including upgrading the included PostgreSQL database from 9.6 (no longer supported) to 14 (stable and supported through 2026). It also changes the way that configuration customizations are made to reduce conflicts with future updates to Central. Please read the following instructions carefully.
+
+.. warning::
+  This upgrade may take more time and disk space than previous updates.
+
+.. warning::
+  Before starting:
+
+  #. :doc:`Back up your server <central-backup>`.
+  #. Make sure you have some time available in case something goes wrong (we recommend at least 3 hours). You may want to announce a maintenance window.
+  #. Review upgrade notes for all versions between your current version and the version you are upgrading to.
+
+#. **Log into the web interface** and make a quick note of some of the data you see, such as submission counts and latest submission dates. You may want to use this information to do a quick spot check after the upgrade is finished.
+
+#. **Log into your server.** If you used our :doc:`DigitalOcean installation steps <central-install-digital-ocean>`, we suggest reviewing the section :ref:`central-install-digital-ocean-build` as a reminder, or if you can't remember your password to start at the top of that section to reset your password.
+
+#. **Go to the central folder.**
+
+   .. code-block:: console
+
+     $ cd central
+
+#. **Upgrade Docker if needed.** Check to see if you have Docker Engine v23.x and Docker Compose v2.16.x or greater:
+
+   .. code-block:: console
+
+     $ docker --version && docker compose version
+
+   If you are using old versions, follow the instructions to install `Docker Engine <https://docs.docker.com/engine/install/ubuntu>`_ (not Desktop) for Ubuntu, the operating system we recommend and support. The instructions will help you setup the Docker ``apt`` repository and install the latest version of Docker Engine and Docker Compose.
+
+#. **Remove docker-compose.** You will be using ``docker compose`` from now on (the dash has been replaced with a space).
+
+   .. code-block:: console
+
+     $ rm -f `which docker-compose`
+
+#. **Migrate configuration customizations.** This will simplify future Central upgrades. First, check what files have been customized:
+
+   .. code-block:: console
+
+     $ git status
+
+   If you do not see any files listed with a `modified:` prefix, you can go on to the next step.
+
+   If you see files listed with a `modified:` prefix, follow instructions for each of them:
+
+   .. dropdown:: ``files/service/config.json.template``
+     :icon: file-code
+
+     #. See what changes have been made:
+
+        .. code-block:: console
+
+          $ git diff files/service/config.json.template
```

Comment:
As I'm going through the steps, I feel very nervous about this step of copying values into another file and deleting my changes to this file long before I know if it will actually work. I think there should be a recommendation to make a copy/backup of this file to refer to just in case.

Reply:
It is also awkward to check my diff and then enter new values into `.env`. I also wish I had the `.env.template` already.

Reply:
I did copy from the blocks, I just felt like I was doing this kind of blindly, not knowing where I'd end up.

I did go seek out the template, and seeing it made me think, "Oh, that's what i'm aiming for, that's not scary". I also used it to copy over the sentry values.

Reply:
I think maybe this is because Cloud customizes everything? For most people only one or maybe two of these blocks will be relevant. Most common will be just email, I think.